internal/clients/recommendations/client.go

```diff
@@ -106,7 +108,10 @@ func (rc *RecommendationsClient) request(method, url string, headers map[string]
 	}
 	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 
-	resp, err := rc.client.Do(req)
+	// TODO context must be passed to the request method
+	req = req.WithContext(context.TODO())
+	doer := strc.NewTracingDoer(rc.client)
+	resp, err := doer.Do(req)
```

Comment:
Shall we split the tracing out to a separate PR here

Reply:
Ah those accidentally got back during conflict rebase, I will drop them.